resources/databaseUser/resource.go

```diff
@@ -0,0 +1,337 @@
+	"github.com/atlas-api-helper/util/configuration"
+	"github.com/atlas-api-helper/util/constants"
+	"github.com/atlas-api-helper/util/validator"
+	"go.mongodb.org/atlas-sdk/v20230201002/admin"
+)
+
+var CreateRequiredFields = []string{constants.Username, constants.Password, constants.PublicKey, constants.PrivateKey, constants.ProjectID}
+var ReadRequiredFields = []string{constants.ProjectID, constants.Username, constants.PublicKey, constants.PrivateKey}
+var DeleteRequiredFields = []string{constants.ProjectID, constants.Username, constants.PublicKey, constants.PrivateKey}
+var ListRequiredFields = []string{constants.ProjectID, constants.PublicKey, constants.PrivateKey, constants.PublicKey, constants.PrivateKey}
+
+func setup() {
+	util.SetupLogger("mongodb-atlas-database-user")
+}
+
+// validateModel to validate inputs to all actions
+func validateModel(fields []string, model *InputModel) error {
+	return validator.ValidateModel(fields, model)
+}
+
+// Create handles the Create event from the Cloudformation service.
+func Create(ctx context.Context, inputModel *InputModel) atlasresponse.AtlasRespone {
+	setup()
+	if errEvent := validateModel(CreateRequiredFields, inputModel); errEvent != nil {
+		util.Warnf(ctx, " create databaseuser is failing with invalid parameters: %#+v", errEvent.Error())
+		return atlasresponse.AtlasRespone{
+			Response:       nil,
+			HttpStatusCode: configuration.GetConfig()[constants.InvalidInputParameter].Code,
+			HttpError:      fmt.Sprintf(configuration.GetConfig()[constants.InvalidInputParameter].Message, errEvent.Error()),
+		}
+	}
+
+	client, peErr := util.NewMongoDBSDKClient(*inputModel.PublicKey, *inputModel.PrivateKey)
+
+	if peErr != nil {
+		util.Warnf(ctx, " Create Mongo client Error: %#+v", peErr.Error())
+		return atlasresponse.AtlasRespone{
+			Response:       nil,
+			HttpStatusCode: configuration.GetConfig()[constants.MongoClientCreationError].Code,
+			HttpError:      configuration.GetConfig()[constants.MongoClientCreationError].Message,
+		}
+	}
+
+	groupID, dbUser := setModel(inputModel)
+	util.Debugf(ctx, "Arguments: Project ID: %s, Request %#+v", groupID, dbUser)
+
+	request := client.DatabaseUsersApi.CreateDatabaseUser(context.Background(), groupID, dbUser)
+
+	databaseUser, _, err := request.Execute()
+
+	if err != nil {
+		fmt.Println("Error creating database user:", err)
+		return atlasresponse.AtlasRespone{
+			Response:       nil,
+			HttpStatusCode: configuration.GetConfig()[constants.UserCreateError].Code,
+			HttpError:      fmt.Sprintf(configuration.GetConfig()[constants.UserCreateError].Message, *inputModel.Username),
+		}
+	}
+	util.Debugf(ctx, "newUser: %+v", databaseUser)
+
+	return atlasresponse.AtlasRespone{
+		Response:       databaseUser,
+		HttpStatusCode: configuration.GetConfig()[constants.UserCreateSuccess].Code,
+		HttpError:      fmt.Sprintf(configuration.GetConfig()[constants.UserCreateSuccess].Message, *inputModel.Username),
+	}
```

Comment:
atlasResponse should be a generic method at least in the same file. Any change in response may lead to many changes in the code.